### PR TITLE
manage server config through .env file

### PR DESCRIPTION
**Pull request by bound2**
Prevent credential leaks & default credential usage by server owners.

`.env` file should be created in `server` folder.

`.env` for `npm run start`
```
DB_HOST=localhost
DB_NAME=snaily-cad
DB_PASSWORD=YOUR_DB_PW_HERE
JWT_SECRET=YOUR_JWT_SECRET_HERE
PROFILE=production
```

`.env` for **Docker**
```
DB_HOST=database
DB_NAME=snaily-cad
DB_PASSWORD=YOUR_DB_PW_HERE
JWT_SECRET=YOUR_JWT_SECRET_HERE
PROFILE=production
```

**Docker** `.env` usage:
`docker-compose --env-file ./server/.env up`

Works straight out of the box in the case of `npm run start` as dotenv node module exists.

--------------------------
This affects existing users who have made their own `config.ts` file (previously in .gitignore).

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -11,7 +11,6 @@ build
 node_modules
 
 # config
-server/config.ts
```
Comment (Dev-CasperTheGhost) [inline .gitignore:14]: Please revert this change, It's safer so it doesn't accidentally gets pushed to git

File: server/config.ts [hunk 2]
```diff
@@ -7,9 +7,9 @@ const config = {
   port: Number(process.env.PORT) || 3030,
   host: process.env.DB_HOST || "localhost",
   user: process.env.DB_USER || "root",
-  password: process.env.DB_PASSWORD || "admin",
+  password: process.env.DB_PASSWORD,
   databaseName: process.env.DB_NAME || "snaily-cad",
-  jwtSecret: process.env.JWT_SECRET || "bongo super cat",
+  jwtSecret: process.env.JWT_SECRET,
   env: process.env.PROFILE || "production",
 };
```
Comment (Dev-CasperTheGhost) [inline server/config.ts:15]: As stated before, please revert this change

**Review by Dev-CasperTheGhost — CHANGES_REQUESTED** [review]
Check request changes, and fix prettier stuff :)

File: .gitignore (hunk 1 above)
Comment (bound2) [reply .gitignore:14]: The idea was to make credential management for both build types (npm & docker) to happen through .env file (in .gitignore). Server config.ts should be committed to the repository as it only takes values from env and user shouldn't modify it directly.

Comment (Dev-CasperTheGhost) [reply .gitignore:14]: I think we should have both, `config.ts` and `.env`, because changing it to .env will break for all the users that have the config.ts. This might create confusion with the current guides too

Comment (bound2) [reply .gitignore:14]: Alright. I'm not that familiar with nodejs, but would it be possible to create abstraction config layer between config.ts and .env. Meaning that the file could be optional and it would read values from there if .env values are missing. I know that TypeScript starts screaming in case file is missing.

Comment (Dev-CasperTheGhost) [reply .gitignore:14]: I don't think that's possible.

**Comment by bound2** [comment]
Going to try and make compatible config. Closing until then.
